### PR TITLE
Manage some basic stuff - early days

### PR DESCRIPTION
**Pull request by djmitche**
This sets up a little bit of infrastructure for managing "projects" within this deployment of TC.  The data at the top would likely move to YAML files eventually.  The use of JSON-e may not be great :)

### CODE REVIEW

File: tc-admin.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+worker_pool_definitions = {
+    'standard-docker-worker': {
+        'description': '',
+        'owner': 'nobody@mozilla.com',
+        'provider_id': 'community-tc-workers',
+	'config': {
+	    "maxCapacity": 5,
+	    "minCapacity": 0,
+	    "capacityPerInstance": 1,
+	    "machineType": "custom-32-29440",
```
Comment (ccooper) [inline tc-admin.py]: Do we want to match this to the default instance type we're using in Firefox CI, i.e. custom 32-36864?

Comment (djmitche) [reply tc-admin.py]: Maybe -- I just copy/pasted this from an existing definition.  Are custom types portable across gcp projects?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,30 @@
...
+```
+
+To apply the configuration, you will need to have a client with suffucient scopes set up in `TASKCLUSTER_CLIENT_ID` and `TASKCLUSTER_ACCESS_TOKEN`.
+Then, run
+
+```
+TASKCLUSTER_ROOT_URL=https://community-tc.services.mozilla.com/. tc-admin apply
+```
+
+Typically a memebr of hte Taskcluster team will do the latter for you once a PR has been merged.
```
Comment (imbstack) [inline README.md]: memebr

File: tc-admin.py (hunk 1 above)
Comment (imbstack) [reply tc-admin.py]: They are portable but this is also a huge machine compared to what people will want most of the time I think. Probably `n1-standard-4` is good enough or something

File: tc-admin.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	    "scheduling": {
+		"onHostMaintenance": "terminate",
+	    },
+	    "userData": {},
+	    "disks": [{
+		"type": "PERSISTENT",
+		"boot": True,
+		"autoDelete": True,
+		"initializeParams": {
+		    "sourceImage": "global/images/taskcluster-worker-googlecompute-2019-09-04t19-01-49z",
```
Comment (imbstack) [inline tc-admin.py]: Let's make this `projects/taskcluster-imaging/global/images/docker-worker-gcp-googlecompute-2019-09-25t19-53-35z` but we should also make sure to make the project this is running in able to access this.

Comment (djmitche) [reply tc-admin.py]: Sounds good.  I assume going forward that we'll adapt to whatever means of producing images we want.  But @milescrabill this is a good example of why having monopacker make a Phabricator request isn't a great solution -- it'd need to know how to make a PR for this repo, too (and be updated when this repo changes its config format).  So ideally monopacker will eventually define a structure for image metadata, and then I'll teach this repo to interpret that.

File: deployment-details.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+All of the information here can be modified via bugs filed in `Operations: Taskcluster Services` in the `Cloud Services` product on https://bugzilla.mozilla.org.
+
+# Worker Manager Providers
+
+* `community-tc-workers` is a google provider corresponding to GCP project `community-tc-workers` (NOTE: this will change to `community-tc-workers-google`)
+
+# GitHub App
+
+The GitHub app to connect your repositories to Taskcluster is is called `Community-TC Integration`, under the Taskcluster organization.
```
Comment (helfi92) [inline deployment-details.md]: is is

File: README.md [hunk 5]
```diff
@@ -1,2 +1,30 @@
...
+See that tool's documentation for background on how the process works.
+
+## Quick Start
+
+Install this app by running `pip install -e .` in this directory.
+
+After making a change the the configuration, you can examine the results with
+
+```
+TASKCLUSTER_ROOT_URL=https://community-tc.services.mozilla.com/. tc-admin diff
```
Comment (helfi92) [inline README.md]: There seems to be a dot at the end of the url. Same thing for L21.

**Comment by djmitche** [comment]
It will, we just don't have a provider configured yet -- https://bugzilla.mozilla.org/show_bug.cgi?id=1584379

File: deployment-details.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Deployment Details
+
+What follows are details about the Community-TC deployment.
+Some of this information is useful for configuring resources in this repository, but lots of this information is handy for folks requesting changes to the deployment.
+
+All of the information here can be modified via bugs filed in `Operations: Taskcluster Services` in the `Cloud Services` product on https://bugzilla.mozilla.org.
```
Comment (petemoore) [inline deployment-details.md]: Not via a pull request against this repository?

I would prefer: "For corrections/improvements to this document, please raise a bug [here](https://url.to.create.a.new.bug)."

File: deployment-details.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+# Worker Manager Providers
+
+* `community-tc-workers` is a google provider corresponding to GCP project `community-tc-workers` (NOTE: this will change to `community-tc-workers-google`)
```
Comment (petemoore) [inline deployment-details.md]: We haven't explained what a google provider is.

File: deployment-details.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+# GitHub App
+
+The GitHub app to connect your repositories to Taskcluster is called `Community-TC Integration`, under the Taskcluster organization.
```
Comment (petemoore) [inline deployment-details.md]: I would prefer something like:

> To enable taskcluster tasks to be run in response to github events (such as a push to a branch, a push to a tag, creation of a github release, etc) you will first need to enable the taskcluster app for your repository. This is required for two reasons:
> 
> 1) It informs the taskcluster community deployment that you wish it to react to events within your repository.
> 2) It grants the taskcluster community deployment the permissions it needs in order to be able to post events to your repository, for example, to comment in your pull requests.
>
> You can enable the integration with the taskcluster community deployment via the `Community-TC Integration` app via [this link](https://github.com/apps/community-tc-integration).

File: README.md [hunk 9]
```diff
@@ -1,2 +1,30 @@
...
+```
+
+To apply the configuration, you will need to have a client with suffucient scopes set up in `TASKCLUSTER_CLIENT_ID` and `TASKCLUSTER_ACCESS_TOKEN`.
+Then, run
+
+```
+TASKCLUSTER_ROOT_URL=https://community-tc.services.mozilla.com tc-admin apply
+```
+
+Typically a member of the Taskcluster team will do the latter for you once a PR has been merged.
```
[the remark this answers is not in the document]
Comment (djmitche) [reply README.md]: We can look into that eventually.  Once we're managing secrets, that gets harder, as we would then have to have all secrets available to a CI task.

File: deployment-details.md (hunk 6 above)
Comment (djmitche) [reply deployment-details.md]: No, not via requests against this repository -- this file details things that are controlled by the configuration of the deployment (stuff that cloudops can change), whereas the repo controls TC resources.

File: README.md [hunk 10]
```diff
@@ -1,2 +1,30 @@
...
+
+Install this app by running `pip install -e .` in this directory.
+
+After making a change the the configuration, you can examine the results with
+
+```
+TASKCLUSTER_ROOT_URL=https://community-tc.services.mozilla.com tc-admin diff
+```
+
+To apply the configuration, you will need to have a client with suffucient scopes set up in `TASKCLUSTER_CLIENT_ID` and `TASKCLUSTER_ACCESS_TOKEN`.
```
[the remark this answers is not in the document]
Comment (djmitche) [reply README.md]: I think it would basically be `*`, with the details depending on the contents of the repository.

**Comment by djmitche** [comment]
> I feel like this guide doesn't really tackle the topics in a systematic way so that a user can read from beginning to end ...

It's not intended to - it's a README in a PR entitled "early days" :)

But you're right, I can definitely be more specific and include more links.  Thanks!

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (djmitche) [reply README.md]: I ended up taking this section out entirely and just focusing on `tc-admin diff`.

File: tc-admin.py [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+import jsone
+
+from tcadmin.appconfig import AppConfig
+from tcadmin.resources import Role, WorkerPool
+
+appconfig = AppConfig()
+
+projects = {
+    'taskcluster': {
+        'admin-roles': ['github-team:taskcluster/core', 'mozilla-group:team_taskcluster'],
```
Comment (milescrabill) [inline tc-admin.py]: We have a `mozilla-group` here which refers to an LDAP group - we only allow Github login in the community deployment

Comment (djmitche) [reply tc-admin.py]: Yeah, I think I had in mind we were using both when I wrote this.  I'll update to use @mozilla/community-ci-admins.

**Comment by djmitche** [comment]
We can update the README if further changes are required :)

**Pull request merged**